torch/lib/THC/THCTensorScatterGather.cu

```diff
@@ -92,7 +92,7 @@ __global__ void THCudaTensor_gatherKernel(
                                                           tensor, &tensorOffset,
                                                           src, &srcOffset);
 
-    IndexType indexValue = (IndexType)index.data[indexOffset] - TH_INDEX_BASE;
+    long indexValue = index.data[indexOffset] - TH_INDEX_BASE;
```

Comment:
Ah no need for any changes. The value will never overflow a long because it's a long tensor.